Diploma project

VladimirPlot opened the pull request:
Реализация первой части дипломного проекта Explore With Me — сервис статистики.

VladimirPlot committed:
Diploma project

Реализация первой части дипломного проекта Explore With Me — сервис статистики.

z8q commented on pom.xml:
```diff
@@ -1,38 +1,93 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 	</properties>
 
+	<dependencies>
```
в корневом и промежуточных pom-ах допустимы блоки dependencyManagement, pluginManagement

определенные dependencies стоит размещать в pom-ах тех саб модулей, которые будут их использовать


отличие dependencyManagement от dependencies в том, что в dependencyManagement мы указываем зависимости с версиями в корневом пом (сами зависимости при этом не применяются ко всем саб-модулям), а dependencies используем в саб-модулях, в них версии можно уже не указывать. Это обеспечивает централизованное управление версиями.

еще один момент с зависимостью spring-boot-starter-parent в блоке parent,
она содержит в себе [spring-boot-dependencies](https://central.sonatype.com/artifact/org.springframework.boot/spring-boot-dependencies/3.3.2), в которой указаны все применяемые в parent зависимости с версиями.
Это позволяет указывать зависимости из parent без версий, например
```xml
<dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
</dependency>
```


про dependencyManagement написал просто для информации, его можно применять только при необходимости, если в проекте нужны какие-нибудь зависимости, которых нет в parent

z8q commented on stats/stats-server/pom.xml:
```diff
@@ -0,0 +1,82 @@
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.32</version>
```
если версию не указать, будет применена версия из [parent](https://central.sonatype.com/artifact/org.springframework.boot/spring-boot-dependencies/3.3.2) 1.18.34

VladimirPlot committed:
Diploma project

Исправил замечания: убрана версия lombok, т.к. она уже указывается в spring-boot-starter-parent.
